### PR TITLE
Send start and end appointment times

### PR DESCRIPTION
**Pull request by Bhavesh-Patel**

### CODE REVIEW

File: compoments/report-repair/repair-availability.js [hunk 1]
```diff
@@ -36,28 +49,28 @@ const RepairAvailability = ({handleChange, values, fromDate}) => {
...
+      const appointmentSlotKey = CreateAppointmentSlotKey(startDateTime, endDateTime)
       const timeString = `${startTime} to ${endTime}`
+      const timeStringSummary = `between ${timeString}`
+      const appointmentSlotData = {timeString, startDateTime:d.startTime, endDateTime:d.endTime, appointmentSlotKey}
       availability[dateString] ?
-        availability[dateString].push(timeString) :
-        availability[dateString] = [timeString]
+        availability[dateString].push(appointmentSlotData) :
+        availability[dateString] = [appointmentSlotData]
+      availabilityValues[appointmentSlotKey] = {startDateTime:d.startTime, endDateTime:d.endTime, display:`${dateString} ${timeStringSummary}`}
```
Comment (Bhavesh-Patel) [inline compoments/report-repair/repair-availability.js:64]: The `availability` map is keyed by `dateString` to allow nesting a days slots within the UI.

The desired state changed by this component needs individual values for start and end times and that's what drove the creation of `availabilityValues`, which is keyed by `appointmentSlotKey`. 

And the `value` property of the radio buttons is also `appointmentSlotKey` allowing lookup upon change.

Happy to change if there is a better and/or simpler way.

File: compoments/report-repair/repair-availability.js [hunk 2]
```diff
@@ -7,8 +7,20 @@ import moment from 'moment';
...
+  const CreateAppointmentSlotKeyFromDateStrings = (startDateTime, endDateTime) => {
+    return CreateAppointmentSlotKey(moment(startDateTime), moment(endDateTime));
+  }
+
+  const CreateAppointmentSlotKey = (startDateTime, endDateTime) => {
+    return `${startDateTime.unix()}-${endDateTime.unix()}`;
+  }
+  let defaultValue;
+  if (values.availability) {
+    defaultValue = CreateAppointmentSlotKeyFromDateStrings(values.availability?.startDateTime, values.availability?.endDateTime)
```
Comment (Bhavesh-Patel) [inline compoments/report-repair/repair-availability.js:19]: The `value` field stores the current selection in a different form than is stored in the state passed in (`values`), hence the need for this conversion. This fixes an issue in selection upon returning to the availability section after making an initial selection.

Perhaps the correct value should be stored in the state and read back in when this component is revisited(?)
If so, then the summary could remove that value prior to submitting as the API doesn't required it.

Comment (Bhavesh-Patel) [reply compoments/report-repair/repair-availability.js]: I've now implemented the above described approach.

File: pages/report-repair/[route].js [hunk 3]
```diff
@@ -59,7 +59,12 @@ function ReportRepair() {
   const [formError, setFormError] = useState();
   const [requestId, setRequestId] = useState();
 
+  const CleanPayload = (payload) => {
```
Comment (elena-vi) [inline pages/report-repair/[route].js:62]:
```suggestion
  const cleanPayload = (payload) => {
```

File: pages/report-repair/[route].js [hunk 4]
```diff
@@ -59,7 +59,12 @@ function ReportRepair() {
...
+    delete payload.availability.appointmentSlotKey
+  }
+
   const submit = (values) => {
+    CleanPayload(values)
```
Comment (elena-vi) [inline pages/report-repair/[route].js:67]:
```suggestion
    cleanPayload(values)
```

**Review by elena-vi — APPROVED** [review]
slight rename would be nice, but otherwise okay

**Comment by Bhavesh-Patel** [comment]
[Integrate repairs API with scheduling API to schedule a repair job](https://trello.com/c/dUu0WkTu/302-integrate-repairs-api-with-scheduling-api-to-schedule-a-repair-job)

**Pull request merged**